lib/reversetunnel/srv.go

```diff
@@ -915,16 +915,16 @@ func (s *server) getRemoteClusters() []*remoteSite {
 func (s *server) GetSite(name string) (RemoteSite, error) {
 	s.RLock()
 	defer s.RUnlock()
-	for i := range s.remoteSites {
-		if s.remoteSites[i].GetName() == name {
-			return s.remoteSites[i], nil
-		}
-	}
 	for i := range s.localSites {
 		if s.localSites[i].GetName() == name {
 			return s.localSites[i], nil
 		}
 	}
+	for i := range s.remoteSites {
```

Comment:
I understand this may fix a test, but rearranging the code to make tests pass doesn't appear like a good solution TBH. Can the test be updated in a way that it doesn't rely on particular order in which clusters are returned? You can use sorting for example.

Reply:
I'd argue changing the order is the correct fix here.

What happened was that when `RouteToCluster` could be empty, the proxy would in that case [pick the first item from the list](https://github.com/gravitational/teleport/blob/8492f62d439fb14dd9ad4059137fdc267f5e0584/lib/srv/regular/proxy.go#L225-L245) returned by `GetSites`. That first item was always a *local* cluster. However, with the non-empty `RouteToCluster` rule in force, `GetSite` started being used in stead of `GetSites` -- but `GetSite` tries to find a match among *remote* clusters first! So if there happen to be two clusters of the same name (as was the case with the test) you would get inconsistent results from `GetSite` and `GetSites` just because of how their internal loops are structured. It's reasonable to expect for the two approaches (loop over `GetSites` looking for a cluster name; pass the cluster name to `GetSite`) to produce equivalent results so I think we should keep the two functions in sync.

That said, the test should be fixed too, in the sense that the two clusters (remote and local) should be disentangled and not use the same name.  Unfortunately I don't have the means to do that at this time.

Reply:
Without this change, which test is failing and on which assertion?

Reply:
```
FAIL: sshserver_test.go:691: SrvSuite.TestProxyReverseTunnel

sshserver_test.go:790:
    s.testClient(c, proxy.Addr(), s.srvAddress, s.srv.Addr(), sshConfig)
sshserver_test.go:646:
    // Request opening TCP connection to the remote host
    c.Assert(se.RequestSubsystem(fmt.Sprintf("proxy:%v", targetAddr)), IsNil)
... value *errors.errorString = &errors.errorString{s:"ssh: subsystem request failed"} ("ssh: subsystem request failed")
```

Reply:
@andrejtokarcik Sorry for delay here. I think I understand now that GetSites iterates over clusters in (local, remote...) order while GetSite - in (remote, local...) order. I agree this inconsistency is unfortunate, however it still feels wrong to just reorder stuff to make the tests pass and have an implicit reliance on the order somewhere, if we want to maintain quality. Would it be hard to update that test? I'm actually not sure how the two clusters, local and remote, can have the same name, I don't think this is a supported use-case?

Reply:
Let's keep this reordering change.
Cluster (site) names should be unique, so the order in `GetSite` doesn't really matter.
Even if you end up with identical site names, preferring local ones makes more sense to me.

Reply:
@r0mant if you're OK with that, please approve the PR